backend/src/views.py

```diff
@@ -5,6 +5,55 @@
 from dateutil.rrule import rrule, MONTHLY, YEARLY, WEEKLY
 import json
 
+from .models import Rule
+from django.views.decorators.csrf import csrf_exempt
+from django.core.exceptions import ObjectDoesNotExist
+from rest_framework import status
+from .serializers import RuleSerializer
+from rest_framework.parsers import JSONParser 
+
+@csrf_exempt
+def update_rule(request, rule_id):
+    rule = Rule.objects.get(id=rule_id)
+    rule_data = JSONParser().parse(request)
+    rule_serializer = RuleSerializer(rule, data=rule_data)
+    if rule_serializer.is_valid():
+        rule_serializer.save()
+        return JsonResponse(rule_serializer.data)
+    return JsonResponse(rule_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+    
+@csrf_exempt
+def get_rule_list(request):
+    rules = Rule.objects.all()
+    rule_serializer = RuleSerializer(rules, many=True)
+    return JsonResponse(rule_serializer.data, safe=False)
+
+def get_rule(request, rule_id):
+    rule = Rule.objects.get(id=rule_id)
+    rule_serializer = RuleSerializer(rule) 
+    return JsonResponse(rule_serializer.data)     
+
+@csrf_exempt
+def create_rule(request):
+    rule_data = JSONParser().parse(request)
+    rule_serializer = RuleSerializer(data=rule_data)
+    if rule_serializer.is_valid():
+        rule_serializer.save()
+        return JsonResponse(rule_serializer.data, status=status.HTTP_201_CREATED) 
+    return JsonResponse(rule_serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+@csrf_exempt
+def delete_rule(request, rule_id):
+    # user = request.user.id
+    try:
+        rule = Rule.objects.get(id=rule_id)
+        rule.delete()
+        return JsonResponse({'message': 'Rule was successfully deleted'}, status=status.HTTP_204_NO_CONTENT)
+    except ObjectDoesNotExist as e:
+        return JsonResponse({'error': str(e)}, safe=False, status=status.HTTP_404_NOT_FOUND)       
```

Comment:
```suggestion
    except ObjectDoesNotExist as e:
        return JsonResponse({'error': 'Rule not found' }, status=status.HTTP_404_NOT_FOUND)       
```
1. safe should not be False.
2. We should not sent stringified exceptions in a response, except in very careful circumstances. See other comment.

Reply:
"other comment": https://github.com/MoneyWise2020/MoneyWise2020/pull/31#discussion_r521113563